include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -798,14 +798,14 @@ struct __simple_write_to_id
     }
 };
 
-template <typename _Predicate>
+template <typename _Predicate, typename _RangeTransform = oneapi::dpl::__internal::__no_op>
 struct __gen_mask
 {
     template <typename _InRng>
     bool
-    operator()(const _InRng& __in_rng, std::size_t __id) const
+    operator()(_InRng&& __in_rng, std::size_t __id) const
     {
-        return __pred(__in_rng[__id]);
+        return __pred((_RangeTransform{}(std::forward<_InRng>(__in_rng)))[__id]);
```

Comment:
For consistency with our functors, how about we add a `_RangeTransform __rng_transform` member after `__pred`?

Reply:
fixed, thanks.